services/catarse/spec/actions/billing/payments/authorize_transaction_spec.rb

```diff
@@ -70,23 +70,82 @@
         end
+      end
+    end
 
-        it 'creates credit card' do
-          result
+    context 'when credit card wasn`t used' do
+      it 'creates credit card' do
+        result
 
-          expect(payment.credit_card.attributes).to include(
-            'user_id' => payment.user_id,
-            'gateway' => payment.gateway,
-            'gateway_id' => gateway_response.dig('card', 'id'),
-            'holder_name' => gateway_response.dig('card', 'holder_name'),
-            'bin' => gateway_response.dig('card', 'first_digits'),
-            'last_digits' => gateway_response.dig('card', 'last_digits'),
-            'country' => gateway_response.dig('card', 'country'),
-            'brand' => gateway_response.dig('card', 'brand'),
-            'expires_on' => Date.parse('2028-02-01')
-          )
-        end
+        expect(payment.credit_card.attributes).to include(
+          'user_id' => payment.user_id,
+          'gateway' => payment.gateway,
+          'gateway_id' => gateway_response.dig('card', 'id'),
+          'holder_name' => gateway_response.dig('card', 'holder_name'),
+          'bin' => gateway_response.dig('card', 'first_digits'),
+          'last_digits' => gateway_response.dig('card', 'last_digits'),
+          'country' => gateway_response.dig('card', 'country'),
+          'brand' => gateway_response.dig('card', 'brand'),
+          'expires_on' => Date.parse('2028-02-01')
+        )
       end
+
+      it 'creates credit card in database' do
+        credit_card_count = Billing::CreditCard.count
+        result
+
+        expect(Billing::CreditCard.count).to eq(credit_card_count + 1)
+      end
+    end
+
+    context 'when payment already has a credit card' do
+      let(:credit_card) { create(:billing_credit_card) }
+
+      before do
+        gateway_response['card']['id'] = credit_card.gateway_id
+        payment.update(credit_card_id: credit_card.id)
+      end
+
+      it 'does not create credit card in database' do
+        credit_card_count = Billing::CreditCard.count
+        result
+
+        expect(Billing::CreditCard.count).to eq(credit_card_count)
+      end
+
+      it 'same credit card for before payment' do
+        result
+
+        expect(payment.credit_card_id).to eq(credit_card.id)
+      end
+    end
+
+    context 'when credit card was used' do
+      let(:credit_card) { create(:billing_credit_card) }
+
+      before do
+        gateway_response['card']['id'] = credit_card.gateway_id
+        allow(Billing::CreditCard).to receive(:find_by).and_return(credit_card)
+      end
+
+      it 'does not create credit card' do
+        credit_card_count = Billing::CreditCard.count
+        result
+
+        expect(Billing::CreditCard.count).to eq(credit_card_count)
+      end
+
+      it 'same credit card for before payment' do
+        result
+
+        expect(payment.credit_card_id).to eq(credit_card.id)
+      end
+    end
```

Comment:
**Problema**
Melhorar legibilidade da mensagen do `it`.

Pra garantir que o result está fazendo o que é pra fazer, tu pode colocar antes dele ou no before um `payment.update!(credit_card_id: nil)`, assim tu garante que o pagamento que está sendo testado não terá cartão de crédito, pois se por algum motivo alguém muda a criação da factory no início do arquivo pra criar o pagamento já com cartão de crédito, esses teus testes podem dar um falso positivo.